core/drivers/clk/sam/at91_master.c

```diff
@@ -179,3 +182,111 @@ const struct clk_master_layout at91sam9x5_master_layout = {
 	.pres_shift = 4,
 	.offset = AT91_PMC_MCKR,
 };
+
+static size_t clk_sama7g5_master_get_parent(struct clk *hw)
+{
+	struct clk_master *master = hw->priv;
+	size_t i = 0;
+
+	for (; i < hw->num_parents; i++)
```

Comment:
`for (i = 0; i < hw->num_parents; i++)` (and please keep the initialization `size_t i = 0;` regardless).